DbService/inc/GrlHeader.hh

```diff
@@ -0,0 +1,39 @@
+#ifndef DbService_GrlHeader_hh
+#define DbService_GrlHeader_hh
+
+// holds a defintion of a goodrun list - name, etc
+// locked==0 is unlocked, locked >0 is locked
+
+#include <iostream>
+#include <string>
+#include <vector>
+
+namespace mu2e {
+
+class GrlHeader {
+ public:
+  typedef std::vector<GrlHeader> HeaderVec;
+  GrlHeader(const std::string& name, int locked = 0, int lid = -1,
+            std::string const& create_time = "",
+            std::string const& create_user = "") :
+      _name(name),
+      _locked(locked), _lid(lid), _create_time(create_time),
+      _create_user(create_user) {}
+  const std::string& name() const { return _name; }
+  int locked() const { return _locked; }
```

Comment:
What is the use case for locking a good run list? Is the implication that grl are 'mutable' unless locked? Does that refer to anything besides extension (ie is actual definition intended to be mutable)?  Who will be allowed to lock/unlock a grl and what mechanism do you forsee for controlling that?

Reply:
The idea is that an unlocked GRL can be extended, but not contracted, for the use case of incoming data, for example "good for pass1" or "good for calibration".   An example for locking a GRL is to fix the list permanently for APS 2026, for example, or 10% e- publication.  There is no unlocking function. (trivial to create new version tho).